src/test/java/io/odpf/firehose/sink/mongodb/MongoSinkFactoryTest.java

```diff
@@ -0,0 +1,156 @@
+package io.odpf.firehose.sink.mongodb;
+
+import com.gojek.de.stencil.client.StencilClient;
+import com.mongodb.ServerAddress;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.metrics.StatsDReporter;
+import io.odpf.firehose.sink.Sink;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class MongoSinkFactoryTest {
+
+    private Map<String, String> configuration;
+
+    @Mock
+    private StatsDReporter statsDReporter;
+
+    @Mock
+    private Instrumentation instrumentation;
+
+    @Mock
+    private StencilClient stencilClient;
+
+    @Before
+    public void setUp() {
+        configuration = new HashMap<>();
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void shouldCreateMongoSink() {
+        configuration.put("SINK_MONGO_CONNECTION_URLS", "localhost:9200 , localhost:9200 ");
+        configuration.put("SINK_MONGO_DB_NAME", "myDb");
+        configuration.put("SINK_MONGO_COLLECTION_NAME", "sampleCollection");
+
+
+        MongoSinkFactory mongoSinkFactory = new MongoSinkFactory();
+        Sink sink = mongoSinkFactory.create(configuration, statsDReporter, stencilClient);
+        assertEquals(MongoSink.class, sink.getClass());
+    }
+
+    @Test
+    public void shouldThrowIllegalArgumentExceptionForEmptyMongoConnectionURLs() {
+        MongoSinkFactory mongoSinkFactory = new MongoSinkFactory();
+        try {
+            mongoSinkFactory.getServerAddresses("", instrumentation);
+        } catch (Exception e) {
+            assertEquals(IllegalArgumentException.class, e.getClass());
+            assertEquals("SINK_MONGO_CONNECTION_URLS is empty or null", e.getMessage());
```

Comment:
Test the exception using rule. Remove try catch.

Reply:
fixed